packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskService.java

```diff
@@ -399,23 +409,31 @@ public void executeTask(TaskInterface task, Bundle data, Error error, TaskExecut
     // The app is not fully loaded as its task manager is not there yet.
     // We need to add event's body to the queue from which events will be executed once the task manager is ready.
     if (!sEventsQueues.containsKey(appId)) {
-      sEventsQueues.put(appId, new ArrayList<Bundle>());
+      sEventsQueues.put(appId, new ArrayList<>());
     }
     sEventsQueues.get(appId).add(body);
 
-    if (!sAppRecords.containsKey(appId)) {
-      // No app record yet - let's spin it up!
+    try {
+      mAppStarter.startApp(mContextRef.get(), new HeadlessAppStarter.Params(appId, task.getAppUrl()), () -> {
+      }, success -> {
+        if (!success) {
+          sEvents.remove(appId);
+          sEventsQueues.remove(appId);
+
+          // Host unreachable? Unregister all tasks for that app.
+          unregisterAllTasksForAppId(appId);
 
-      if (!loadApp(appId, task.getAppUrl())) {
-        // Loading failed because parameters are invalid - unregister the task.
-        try {
-          unregisterTask(task.getName(), appId, null);
-        } catch (Exception e) {
-          Log.e(TAG, "Error occurred while unregistering invalid task.", e);
         }
-        appEvents.remove(eventId);
-        sEventsQueues.remove(appId);
+      });
+    } catch (IllegalArgumentException ignored) {
```

Comment:
`startApp` can also throw `IllegalStateException`

Reply:
Changed to newly created AppConfigurationError. This is checked exception, which better suits our needs.